params_test.go

```diff
@@ -20,3 +22,66 @@ func TestFormatTime(test *testing.T) {
 	formattedTime := formatTime(time)
 	t.Assert(formattedTime, formatTime(time), "formatTime()")
 }
+
+func TestParamsWithOptions(test *testing.T) {
+	t := &T{test}
+
+	ctx := context.Background()
+	iKey := "asdf"
+	p := &Params{}
+	p.ApplyOptions([]Option{
+		WithIdempotencyKey(iKey),
+		WithContext(ctx),
+	})
+
+	t.Assert(iKey, p.IdempotencyKey, "ApplyOptions()")
+	t.Assert(ctx, p.Context, "ApplyOptions()")
+}
+
+func TestCommonParamsWithOptions(test *testing.T) {
+	t := &T{test}
+
+	ctx := context.Background()
+	iKey := "asdf"
+	c := CommonParamsWithOptions([]Option{
+		WithIdempotencyKey(iKey),
+		WithContext(ctx),
+	})
+
+	t.Assert(iKey, c.Params.IdempotencyKey, "CommonParamsWithOptions()")
+	t.Assert(ctx, c.Params.Context, "CommonParamsWithOptions()")
+}
+
+func TestWithIdempotencyKey(test *testing.T) {
+	t := &T{test}
+
+	key := "asdf"
+	p := &Params{}
+	opt := WithIdempotencyKey(key)
+	opt(p)
```

Comment:
on these tests, would be worth checking the values that are set in p to make sure they are as-expected.